src/site/en/blog/2020-07-09-no-to-routers/index.md

```diff
@@ -0,0 +1,246 @@
+
+No, it's not. It could be, but only in simple hello-world-ish cases, which get complex as soon as your app becomes 1 month old.
+
+In general it is a fully-fledged process with dependency loading, data processing and decision making. It is also full of side-effects: from external dependencies and data loading to browser history management.
+
+### Mistake #3: immediate transitions
+
+Let's assume, you are on a simple website with no javascript at all. When you click the link, are you immediately transitioned to your destination? No, even in this simple case you have to wait until the next page is loaded.
+
+Waiting for transitions is in the DNA of the web from day one. Users got used to waiting after clicking the link, they **expect the next page to be loaded.** This means, it's ok to wait, because it is going to show me my entire friend list, and that's a lot of data, and I'm on 2G internet in the middle of nowhere, so I totally understand.
+
+Router should allow to transition out of page, handle waiting time, then transition to the next page. That's what browsers already do with websites, and the least we can do is not break it.
+
+### Mistake #4: no place for dependencies or common behaviour
+
+<figure>
+    <img
+        src="ever-ready.jpg"
+        alt='"Ever-ready" by Anna Zarubey'
+        srcset="ever-ready@2x.jpg 2x"
+        width="1200"
+        height="675"
+    />
+    <figcaption>"Ever-ready" by <a href="https://t.me/anna_zarubey">Anna Zarubey</a></figcaption>
+</figure>
+
+This is generally a combination of #2 and #3, but feels like something to be addressed explicitly.
+
+Routers tend to work with pages as if the app already has everything it needs to display every page. And that may be true for calculator, or some mini game.
+
+In reality we have network-heavy applications, which require both data loading and a lot of javascript and styles to display it. And most of the users won't even visit that one heavy page. So the most logical solution is to separate its resources from the rest of the app.
+
+Now if we are going to separate that page, we have to put all the preconditions and dependencies inside of the page itself, but is it really where they belong? Also: do you enjoy repeating "if no user is authorised, redirect to login" for every page?
+
+### Mistake #5: one and only one path for every page
+
+<figure>
+    <img
+        src="next-room-is-outside.jpg"
+        alt='"Next room is outside" by Anna Zarubey'
+        srcset="next-room-is-outside@2x.jpg 2x"
+        width="1200"
+        height="675"
+    />
+    <figcaption>"Next room is outside" by <a href="https://t.me/anna_zarubey">Anna Zarubey</a></figcaption>
+</figure>
+
+There is a stable <abbr title="User experience">UX</abbr> trend to split long forms into multiple steps called "form wizards". If you are aiming for the good UX, you would usually prefer to add those steps to browser history. This will allow user to navigate between form steps with browser buttons.
```

Comment:
```suggestion
There is a stable <abbr title="User experience">UX</abbr> trend to split long forms into multiple steps called "form wizards". If you aim for the good UX, you would usually prefer to add those steps to browser history. It will allow the user to navigate between form steps with browser buttons.
```